src/structure/model.jl

```diff
@@ -0,0 +1,1089 @@
+name read from the PDB file.
+"""
+defaultresidue(disordered_res::DisorderedResidue) = disordered_res.names[defaultresname(disordered_res)]
+
+"""
+Get the residue names in a `DisorderedResidue`. Default residue name is placed
+first, then others are ordered alphabetically.
+"""
+resnames(disordered_res::DisorderedResidue) = sort(collect(keys(disordered_res.names)), lt= (res_name_one, res_name_two) -> (isless(res_name_one, res_name_two) && res_name_two != defaultresname(disordered_res)) || res_name_one == defaultresname(disordered_res))
+
+resname(disordered_res::DisorderedResidue) = defaultresname(disordered_res)
+chainid(disordered_res::DisorderedResidue) = chainid(defaultresidue(disordered_res))
+resnumber(disordered_res::DisorderedResidue) = resnumber(defaultresidue(disordered_res))
+inscode(disordered_res::DisorderedResidue) = inscode(defaultresidue(disordered_res))
+ishetres(disordered_res::DisorderedResidue) = ishetres(defaultresidue(disordered_res))
+atomnames(disordered_res::DisorderedResidue) = atomnames(defaultresidue(disordered_res))
+atoms(disordered_res::DisorderedResidue) = atoms(defaultresidue(disordered_res))
+
+isdisorderedres(::DisorderedResidue) = true
+
+# Constructor acts as a setter for the default residue name
+function DisorderedResidue(disordered_res::DisorderedResidue, default::ASCIIString)
+    @assert default in resnames(disordered_res) "The new default residue name must be present in the residue"
+    return DisorderedResidue(disordered_res.names, default)
+end
+
+
+# Chain getters/setters
+chainid(chain::Chain) = chain.id
+
+"Get the sorted list of `AbstractResidue`s in a `Chain`."
+resids(chain::Chain) = chain.res_list
+
+"Access the dictionary of `AbstractResidue`s in a `Chain`."
+residues(chain::Chain) = chain.residues
+
+
+# Model getters/setters
+
+"Get the model number of a `Model`."
+modelnumber(model::Model) = model.number
+
+"Get the sorted chain IDs of the chains in a `Model` or `ProteinStructure`."
+chainids(model::Model) = sortchainids(collect(keys(model.chains)))
+
+"Access the dictionary of `Chain`s in a `Model`."
+chains(model::Model) = model.chains
+
+
+# ProteinStructure getters/setters
+
+"Get the name of a `ProteinStructure`."
+structurename(struc::ProteinStructure) = struc.name
+
+"Get the sorted `Model` numbers from a `ProteinStructure`."
+modelnumbers(struc::ProteinStructure) = sort(collect(keys(struc.models)))
+
+"Access the dictionary of `Model`s in a `ProteinStructure`."
+models(struc::ProteinStructure) = struc.models
+
+"""
+Get the default `Model` in a `ProteinStructure`. This is the `Model` with the
+lowest model number.
+"""
+defaultmodel(struc::ProteinStructure) = struc.models[modelnumbers(struc)[1]]
+
+chainids(struc::ProteinStructure) = countmodels(struc) > 0 ? chainids(defaultmodel(struc)) : Char[]
+
+
+# Sorting functions
+
+# Sort atoms by serial
+Base.sort!{T <: AbstractAtom}(atoms::Vector{T}) = sort!(atoms, by=serial)
```

Comment:
It is not a good manner to override the `sort!` method because people will not expect `sort!` for `Vector` is overridden. Moreover, if one passes `rev=true` to sort atoms, this will result in unexpected result. The best way to sort elements by type-specific order is defining `Base.isless` method for that type. So, add `Base.isless(a1::AbstractAtom, a2::AbstractAtom) = isless(serial(a1), serial(a2))` somewhere.